platform/linux-generic/include/odp/api/plat/event_inlines.h

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2018, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_PLAT_EVENT_INLINES_H_
+#define ODP_PLAT_EVENT_INLINES_H_
+
+#include <odp/api/abi/buffer.h>
+#include <odp/api/plat/buffer_inline_types.h>
+
+/** @cond _ODP_HIDE_FROM_DOXYGEN_ */
+
+extern const _odp_buffer_inline_offset_t _odp_buffer_inline_offset;
+
+#ifndef _ODP_NO_INLINE
+	/* Inline functions by default */
+	#define _ODP_INLINE static inline
+	#define odp_event_type __odp_event_type
```

Comment:
It might be better to put this `#define` right before the function.

Reply:
Actually, I'd expect this to be in a common file rather than in each `xxx_inline.h` file. Presumably it's here because that was simpler to illustrate the alternate pattern given that only events are using this structure for now.

Reply:
I think it's cleaner to have on one ifdef in the file and gather all function renames under that. It's hard to read a file with N functions and N ifdefs scattered around the file (a ifdef before each function).

Bill, these rename defines are file specific (no common rename header). Also it does not make this file simpler or more readable, if _ODP_INLINE would be defined in another header file and then that file would be included here - one line define vs one line include. Downside of an include is that the reader of this file would need to go and check that file also and cross check an ifdef there.